server/sonar-server/src/main/java/org/sonar/server/usergroups/ws/UpdateAction.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.usergroups.ws;
+
+import java.util.Arrays;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService.NewAction;
+import org.sonar.api.server.ws.WebService.NewController;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.user.GroupDto;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.user.UserSession;
+
+import static org.sonar.core.persistence.MyBatis.closeQuietly;
+
+public class UpdateAction extends AbstractGroupUpdate implements UserGroupsWsAction {
+
+  public UpdateAction(DbClient dbClient, UserSession userSession) {
+    super(dbClient, userSession);
+  }
+
+  @Override
+  public void define(NewController context) {
+    NewAction action = context.createAction("update")
+      .setDescription("update a group.")
+      .setHandler(this)
+      .setPost(true)
+      .setResponseExample(getClass().getResource("example-update.json"))
+      .setSince("5.2");
+
+    action.createParam(PARAM_ID)
+      .setDescription("Identifier of the group.")
+      .setExampleValue("42")
+      .setRequired(true);
+
+    action.createParam(PARAM_NAME)
+      .setDescription(String.format("New name for the group. A group name cannot be larger than %d characters and must be unique. " +
+        "The value 'anyone' (whatever the case) is reserved and cannot be used.", NAME_MAX_LENGTH))
+      .setExampleValue("sonar-users");
+
+    action.createParam(PARAM_DESCRIPTION)
+      .setDescription(String.format("New description for the group. A group description cannot be larger than %d characters.", DESCRIPTION_MAX_LENGTH))
+      .setExampleValue("Default group for new users");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    userSession.checkLoggedIn().checkGlobalPermission(GlobalPermissions.SYSTEM_ADMIN);
+
+    Long groupId = request.mandatoryParamAsLong(PARAM_ID);
```

Comment:
It's a string in WS response, but on DB side it's really a long. So I prefer to validate this as early as possible.